service/matching/taskQueueManager.go

```diff
@@ -330,6 +330,16 @@ func (c *taskQueueManagerImpl) Stop() {
 	c.unloadFromEngine()
 }
 
+// isVersioned returns true if this is a tqm for a "versioned [low-level] task queue". Note
+// that this is a different concept from the overall [high-level] task queue having versioning
+// data associated with it, which is the usual meaning of "versioned task queue". In this case,
+// it means whether this is a tqm processing a specific version set id. Unlike non-root
+// partitions which are known (at some level) by other services, [low-level] task queues with a
+// version set should not be interacted with outside of the matching service.
```

Comment:
This is maybe better, feel free to ignore it. Perhaps consider renaming the function to `managesSpecificVersionSet` or something?

```suggestion
// isVersioned returns true if this is a tqm for a specific version set in the 
// build-id-based versioning feature. It does not return true for tqms 
// which represent a partition of a task queue as a user might see it, as
// such task queues may map to more than one version set.
```

Reply:
@dnr

Reply:
I like the first sentence better but not the second. "as a user might see it" doesn't really mean anything to me. Users don't see partitions either :man_shrugging:  And to be very pedantic, a "versioned tqm" isn't 1:1 with a version set either, in rare cases a version set might have multiple set ids so it'll need multiple tqms (if there are spooled tasks)